src/browser/components/relate-api/relate-api.utils.js

```diff
@@ -0,0 +1,236 @@
+import { getMainDefinition } from 'apollo-utilities'
+import { onError } from 'apollo-link-error'
+import { NATIVE, KERBEROS } from 'services/bolt/boltHelpers'
+import {
+  SWITCH_CONNECTION,
+  SWITCH_CONNECTION_FAILED
+} from 'shared/modules/connections/connectionsDuck'
+import { INJECTED_DISCOVERY } from 'shared/modules/discovery/discoveryDuck'
+
+export const createClient = (apiEndpoint, apiClientId = null) => {
+  const apiEndpointUrl = new URL(apiEndpoint)
+  const apiEndpointNoScheme = `${apiEndpointUrl.host}${
+    apiEndpointUrl.pathname ? apiEndpointUrl.pathname : ''
+  }`
+
+  const httpLink = createHttpLink({
+    uri: `http://${apiEndpointNoScheme}`
+  })
+
+  const wsLink = new WebSocketLink({
+    uri: `ws://${apiEndpointNoScheme}`,
+    options: {
+      reconnect: true,
+      connectionParams: {
+        ClientId: apiClientId
+      }
+    }
+  })
+
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        ClientId: apiClientId
+      }
+    }
+  })
+
+  const link = split(
+    // split based on operation type
+    ({ query }) => {
+      const { kind, operation } = getMainDefinition(query)
+      return kind === 'OperationDefinition' && operation === 'subscription'
+    },
+    wsLink,
+    authLink.concat(httpLink)
+  )
+
+  const errorLink = onError(({ graphQLErrors, networkError }) => {
+    if (graphQLErrors) {
+      graphQLErrors.forEach(({ message, locations, path }) =>
+        console.log(
+          `Relate API GraphQL error: Message: ${message}, Location: ${locations}, Path: ${path}`
+        )
+      )
+    }
+
+    if (networkError) {
+      console.log(`Relate API Network error: ${networkError}`)
+    }
+  })
+
+  const client = new ApolloClient({
+    link: errorLink.concat(link),
+    cache: new InMemoryCache()
+  })
+
+  return client
+}
+
+export function getPrefersColorScheme (workspaceData) {
+  return (
+    ((workspaceData.workspace || workspaceData.onWorkspaceChange).host || {})
+      .prefersColorScheme || null
+  )
+}
+
+const graphStatus = {
+  ACTIVE: 'ACTIVE'
+}
+export function getActiveGraphData (workspaceData) {
+  return (
+    (workspaceData &&
+      workspaceData.workspace &&
+      workspaceData.workspace.projects) ||
+    []
+  ).reduce((activeGraph, project) => {
+    if (!project.graphs) {
+      return activeGraph
+    }
+    const active = project.graphs.filter(
+      graph => graph.status === graphStatus.ACTIVE
+    )
+    if (!active || !active.length) {
+      return activeGraph
+    }
+    return active[0]
+  }, null)
+}
+
+export const getCredentialsForGraph = (protocol, graph) => {
+  if (!graph || !graph.connection) {
+    return null
+  }
+  const { principals = null } = graph.connection
+  if (!principals) {
+    return null
+  }
+  if (!principals.protocols) {
+    return null
+  }
+  if (typeof principals.protocols[protocol] === 'undefined') {
+    return null
+  }
```

Comment:
these if statements could be simplified with lodash [_.has()](https://lodash.com/docs/4.17.15#has) 
```js
if (_has(graph, `connection.principals.protocols[${protocol}]`)) ...
```